plugins/orchestrator/src/components/WorkflowDefinitionsListComponent/WorkflowDefinitionsListComponent.tsx

```diff
@@ -27,26 +30,78 @@ type WorkflowsTableProps = {
 
 export const WorkflowsTable = ({ items }: WorkflowsTableProps) => {
   const orchestratorApi = useApi(orchestratorApiRef);
-
   const navigate = useNavigate();
   const definitionLink = useRouteRef(workflowDefinitionsRouteRef);
   const executeWorkflowLink = useRouteRef(executeWorkflowRouteRef);
   const editLink = useRouteRef(editWorkflowRouteRef);
+  const [data, setData] = useState<Row[]>([]);
 
   interface Row {
     id: string;
     name: string;
+    lastRun: string;
+    lastRunStatus: string;
+    type: WorkflowType;
+    components: string;
     format: string;
   }
 
-  const columns: TableColumn[] = [{ title: 'Name', field: 'name' }];
-  const data: Row[] = items.map(item => {
-    return {
-      id: item.definition.id,
-      name: item.definition.name ?? '',
-      format: extractWorkflowFormatFromUri(item.uri),
-    };
-  });
+  const columns: TableColumn[] = [
+    { title: 'Name', field: 'name' },
+    { title: 'Last run', field: 'lastRun' },
+    { title: 'Last run status', field: 'lastRunStatus' },
+    { title: 'Type', field: 'type' },
+    { title: 'Components', field: 'components' },
+  ];
+
+  const getInitialState = useCallback(() => {
+    return items.map(item => {
+      return {
+        id: item.definition.id,
+        name: item.definition.name ?? '',
+        lastRun: '',
+        lastRunStatus: '',
+        type: item.definition.annotations?.find(
+          annotation => annotation === ASSESSMENT_WORKFLOW_TYPE,
+        )
+          ? WorkflowType.ASSESSMENT
+          : WorkflowType.INFRASTRUCTURE,
+        components: '---',
+        format: extractWorkflowFormatFromUri(item.uri),
+      };
+    });
+  }, [items]);
+
+  const load = useCallback(
+    (initData: Row[]) => {
+      let clonedData: Row[] = [];
+      orchestratorApi.getInstances().then(instances => {
+        for (const init_row of initData) {
+          const row = { ...init_row };
+          const instancesById = instances.filter(
+            instance => instance.processId === row.id,
+          );
+          const lastRunInstance = instancesById.at(-1);
+          if (lastRunInstance) {
+            row.lastRun = moment(lastRunInstance.start?.toString()).format(
+              'MMMM DD, YYYY',
+            );
+            row.lastRunStatus = lastRunInstance.state;
+            row.components = instancesById.length.toString();
+          }
+          clonedData = [...(clonedData || []), row];
```

Comment:
Two things:
- The `clonedData` are never false-like since they are always initialized above
- `clonedData.push(row)` can be more handy here

Reply:
good suggestion!